io/zenoh-transport/tests/unicast_transport.rs

```diff
@@ -839,5 +872,398 @@ tOzot3pwe+3SJtpk90xAQrABEO0Zh2unrC8i83ySfg==
+            reliability: Reliability::BestEffort,
+        },
+    ];
+    // Run
+    let endpoints = vec![client_endpoint];
+    let result = task::block_on(run(
+        &endpoints,
+        Some(&server_endpoint),
+        &channel,
+        &MSG_SIZE_ALL,
+    ));
+    assert!(result.is_err());
+    let err = result.unwrap_err();
+    assert!(err
+        .to_string()
+        .contains(RUSTLS_CERTIFICATE_REQUIRED_ALERT_DESCRIPTION));
+}
+
+#[cfg(all(feature = "transport_tls", target_family = "unix"))]
+#[test]
+fn transport_unicast_tls_two_way_auth_wrong_certs_fail() {
+    use zenoh_link::tls::config::*;
+
+    task::block_on(async {
+        zasync_executor_init!();
+    });
+
+    let client_auth = "true";
+
+    // Define the locator
+    let mut client_endpoint: EndPoint = ("tls/localhost:10463").parse().unwrap();
+    client_endpoint.extend_configuration(
+        [
+            (TLS_ROOT_CA_CERTIFICATE_RAW, CLIENT_CA),
+            // Using the SERVER_CERT and SERVER_KEY in the client to simulate the case the client has
+            // wrong certificates and keys. The SERVER_CA (cetificate authority) will not recognize
+            // these certificates as it is expecting to receive CLIENT_CERT and CLIENT_KEY from the
+            // client.
+            (TLS_CLIENT_CERTIFICATE_RAW, SERVER_CERT),
+            (TLS_CLIENT_PRIVATE_KEY_RAW, SERVER_KEY),
+            (TLS_CLIENT_AUTH, client_auth),
+        ]
+        .iter()
+        .map(|(k, v)| ((*k).to_owned(), (*v).to_owned())),
+    );
+
+    // Define the locator
+    let mut server_endpoint: EndPoint = ("tls/localhost:10463").parse().unwrap();
+    server_endpoint.extend_configuration(
+        [
+            (TLS_ROOT_CA_CERTIFICATE_RAW, SERVER_CA),
+            (TLS_SERVER_CERTIFICATE_RAW, SERVER_CERT),
+            (TLS_SERVER_PRIVATE_KEY_RAW, SERVER_KEY),
+            (TLS_CLIENT_AUTH, client_auth),
+        ]
+        .iter()
+        .map(|(k, v)| ((*k).to_owned(), (*v).to_owned())),
+    );
+    // Define the reliability and congestion control
+    let channel = [
+        Channel {
+            priority: Priority::default(),
+            reliability: Reliability::Reliable,
+        },
+        Channel {
+            priority: Priority::default(),
+            reliability: Reliability::BestEffort,
+        },
+        Channel {
+            priority: Priority::RealTime,
+            reliability: Reliability::Reliable,
+        },
+        Channel {
+            priority: Priority::RealTime,
+            reliability: Reliability::BestEffort,
+        },
+    ];
+    // Run
+    let endpoints = vec![client_endpoint];
+    let result = task::block_on(run(
+        &endpoints,
+        Some(&server_endpoint),
+        &channel,
+        &MSG_SIZE_ALL,
+    ));
+    assert!(result.is_err());
+    let err = result.unwrap_err();
+    assert!(err
+        .to_string()
+        .contains(RUSTLS_HANDSHAKE_FAILURE_ALERT_DESCRIPTION));
+}
+
+//*************************************/
```

Comment:
Since you are adding this sections, shouldn't we remove the in-line certificates from the `tls` and `quic` tests and only refers to these ones?